unicodecomplete.py

```diff
@@ -44,37 +44,77 @@ def enabled(name, default=True):
     return get_settings().get(name, default)
 
 
-def can_convert(view):
+def replacement(m, instant=False):
+    """
+    Returns the conversion for regex match m (with groups 'symbol', 'prefix'
+    and 'chars'), None if no conversion is possible.
+    If instant=True, restricts to conversions suitable for instant insertion
+    """
+    symbol = m.groupdict().get('symbol')
+    prefix = m.groupdict().get('prefix')
+    chars = m.groupdict().get('chars')
+
+    if symbol is not None:
+        # Accept explicit symbol names; in instant mode, refuse \^... and \_...
+        # (which are left to subs/supers) and ambigous prefixes
+        rep = symbol_by_name(symbol)
+        if rep and (not instant or (symbol[0] not in "^_" and symbol_by_prefix(symbol, unique=True))):
+            return rep
+
+        # Convert unambiguous prefixes
+        if enabled('accept_prefixes'):
+            rep = symbol_by_prefix(symbol, unique=True)
+            if rep:
+                return rep
+
+        # Convert subscript and superscripts, but not in instant mode (it would
+        # convert immediately at \^ or \_)
+        if enabled('convert_sub_super') and is_script(symbol) and (not instant or symbol.endswith(" ")):
+            script_char, chars = get_script(symbol.strip())
+            reps = [symbol_by_name(script_char + ch) for ch in chars]
+            if all(reps):
+                return ''.join(reps)
+
+        # Convert Unicode codes
+        if enabled('convert_codes'):
+            if symbol_by_code(u'\\' + symbol):
+                return True
+
+        # In instant mode, accept symbols when followed by an invalid character.
+        # For instance, when typing "x" in "\alphax", recognize that "\alpha"
```

Comment:
Looks a bit hacky, but maybe it's ok.
If we type 'x' after '\alpha' - it will convert '\alpha'
Will it convert if we _paste_ 'x' after '\alpha' with Ctrl+V? Looks like it will
But it won't - if we paste 'xyz' after '\alpha', because prefix will be '\alphaxy' (without last 'z')
It is expected?

Reply:
It is the intended behavior, though we could discuss how satisfying it is. Ideally you'd want to behave as if all characters were typed in sequence, but testing that doesn't seem too straightforward and I'm afraid of (1) performance issues, as pasting tens of thousands of characters is quite common, and (2) unduly substituting stuff that should've been left verbatim when pasting chunks of code. I think it's fair to have a typing-oriented input mode rather than a text-oriented input mode for instant mode since it is interactive by design.